contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -1820,6 +1821,22 @@ check_alter_role_stmt(GrantRoleStmt *stmt)
 				(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
 				 errmsg("Current login %s does not have permission to alter role %s",
 						GetUserNameFromId(GetSessionUserId(), true), granted_name)));
+	
+	/* Get the database owner. */
+	db_owner_name = get_owner_of_db(get_cur_db_name());
+
+	/*
+	 * Disallow ALTER ROLE if current login is not a member of sysadmin and
+	 * not the database owner.
+	 */
+	if (!has_privs_of_role(GetSessionUserId(), get_role_oid("sysadmin", false)) &&
+		!has_privs_of_role(GetSessionUserId(),get_role_oid(db_owner_name, false)))
```

Comment:
Was this a bug we are fixing?

Reply:
No, this is related to these database roles. I didn't add another check for db roles since all the current tests were working as expected. We might have to bypass all checks for `db_securityadmin` since its member can ALTER ROLE.